.gitignore

```
@@ -0,0 +1,4 @@
+.DS_Store
+.vscode/
+composer.lock
+vendor/
```

```suggestion
/vendor
```

This just ensures that we only ignore the vendor directory at the root, and not any vendor directory that dependencies might have themselves.